.github/workflows/e2e.generic.push.branch1.default.slsa3.yml

```diff
@@ -0,0 +1,128 @@
+name: Generic push branch1 default SLSA2
+
+on:
+  schedule:
+    - cron: "0 3 * * *"
+  workflow_dispatch:
+  push:
+    branches: [main, branch1]
+
+permissions: read-all
+
+env:
+  GH_TOKEN: ${{ secrets.E2E_GO_TOKEN }}
+  ISSUE_REPOSITORY: slsa-framework/slsa-github-generator
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'main' && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch')
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+      - run: ./.github/workflows/scripts/e2e-push.sh
+
+  shim:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+    outputs:
+      continue: ${{ steps.verify.outputs.continue }}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+      - id: verify
+        run: |
+          set -euo pipefail
+
+          source "./.github/workflows/scripts/e2e-utils.sh"
+
+          THIS_FILE=$(e2e_this_file)
+          BRANCH=$(echo "$THIS_FILE" | cut -d '.' -f4)
+          if [[ "$BRANCH" == "${{ github.ref_name }}" ]]; then
```

Comment:
in general, use GITHUB_REF_NAME to avoid script injections. (not a problem here, though)

Reply:
Yeah, This is mostly copied from previous code. I'll see I can update the lot.